src/ray/util/type_traits.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+namespace ray {
+
+template <typename T>
+constexpr bool AlwaysFalse = false;
+
+template <typename T>
+constexpr bool AlwaysTrue = true;
+
+template <int N>
+constexpr bool AlwaysFalseValue = false;
+
+template <int N>
+constexpr bool AlwaysTrueValue = true;
```

Comment:
some of these are not used?

Reply:
yeah we only used 1, left this for completeness. do you want me to remove all unused ones?

Reply:
Not a big deal. Up to you.